src/native/libs/System.Native/pal_signal.c

```diff
@@ -313,13 +314,10 @@ static void* SignalHandlerLoop(void* arg)
     free(arg);
     assert(pipeFd >= 0);
 
-    char* threadName = ".NET SigHandler";
-#if defined(__linux__) || defined(__FreeBSD__)
-    pthread_setname_np(pthread_self(), threadName);
-#endif
-#if defined(__APPLE__)
-    pthread_setname_np(threadName);
-#endif
+    if (minipal_set_thread_name(pthread_self(), ".NET SigHandler") != 0)
```

Comment:
Being unable to set a name seems like the wrong thing to return on. We should assert we can set the name, but I don't think it is appropriate to consider that a failure. Is there some reason to consider an inability to set the thread name, arguably simply a diagnostic nicety, a bigger failure?

Reply:
No copied from coreclr. Mono and libs don't care about the result, coreclr does. Is there a reason why we should or shouldn't align the usage behavior?

Reply:
Nope, copied from coreclr. Mono and libs don't care about the result, coreclr does. Is there a reason why we should or shouldn't align the usage styles?

Reply:
I don't think so. My preference would be to not fail on this sort of thing. Asserting makes good sense, but failure seems wrong.